cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/LineageOperationsProcessor.java

```diff
@@ -189,9 +189,12 @@ private void addMergeOperation(Set<String> stageInputs,
         // in case if rename is done on the join keys. In this case Join operation will read fields of the form
         // <stagename.fieldname> however rename operations followed by it will simply read the output of the
         // Join which wont be tagged with stagename.
+
+        //TODO CDAP-13298 Field names and stage names can have '.' in them.
+        // Probably better to support new JOIN operation type which takes stage name and field name.
         Iterable<String> stageFieldPair = Splitter.on(".").omitEmptyStrings().trimResults().split(field);
-        if (Iterables.size(stageFieldPair) > 1) {
-          String stageName = Iterables.get(stageFieldPair, 0);
+        String stageName = Iterables.get(stageFieldPair, 0);
+        if (Iterables.size(stageFieldPair) > 1 && stageOperations.keySet().contains(stageName)) {
```

Comment:
this check doesn't help because the field has already been split on '.' I checked the Joiner plugin and it looks like the Joiner plugin has the same issue like you mentioned. Let's mark this as TODO and fix it in 5.0.1, seeing how the only plugin we have has the same problem.

Reply:
Added TODO on line 193.